src/languages/vbnet.js

```diff
@@ -33,30 +53,18 @@ export default function(hljs) {
     illegal: '//|{|}|endif|gosub|variant|wend|^\\$ ', /* reserved deprecated keywords */
     contains: [
       hljs.inherit(hljs.QUOTE_STRING_MODE, {contains: [{begin: '""'}]}),
-      hljs.COMMENT(
-        '\'',
-        '$',
-        {
-          returnBegin: true,
-          contains: [
-            {
-              className: 'doctag',
-              begin: '\'\'\'|<!--|-->',
-              contains: [hljs.PHRASAL_WORDS_MODE]
-            },
-            {
-              className: 'doctag',
-              begin: '</?', end: '>',
-              contains: [hljs.PHRASAL_WORDS_MODE]
-            }
-          ]
-        }
-      ),
+      commentMode,
       hljs.C_NUMBER_MODE,
       {
         className: 'meta',
-        begin: '#', end: '$',
-        keywords: {'meta-keyword': 'if else elseif end region externalsource'}
+        begin: '#(?:const|disable|else|elseif|enable|end|externalsource|if|region)\\b', end: '$',
+        keywords: {'meta-keyword': 'const disable else elseif enable end externalsource if region'},
+        contains: [commentMode]
+      },
+      {
+        className: 'literal',
+        begin: '#', end: '#',
+        illegal: '\\n'
```

Comment:
```suggestion
        illegal: /\n/
```

Prefer regex when you can use it.